File: app/src/main/java/starbright/com/projectegg/features/recipelist/RecipeListContract.java
package starbright.com.projectegg.features.recipelist;

import java.util.List;

import starbright.com.projectegg.data.local.model.Ingredient;
import starbright.com.projectegg.data.local.model.Recipe;
import starbright.com.projectegg.features.base.BasePresenter;
import starbright.com.projectegg.features.base.BaseView;

class RecipeListContract {

    interface View extends BaseView<Presenter> {
        void setupRecyclerView();

        void setupSwipeRefreshLayout();

        void showLoadingBar();
        void hideLoadingBar();

        void bindRecipesToList(List<Recipe> recipes);
        void showDetail(String recipeId);

        void showErrorSnackBar(String errorMessage);
    }

    interface Presenter extends BasePresenter{
        void handleListItemClicked(int position);
        void handleRefresh();

        void setIngredients(List<Ingredient> ingredients);
    }
}

File: app/src/main/java/starbright/com/projectegg/features/recipelist/RecipeListContract.kt
/**
 * Created by Andreas on 22/9/2018.
 */

package starbright.com.projectegg.features.recipelist

import starbright.com.projectegg.data.RecipeConfig
import starbright.com.projectegg.data.model.Recipe
import starbright.com.projectegg.features.base.BasePresenterContract
import starbright.com.projectegg.features.base.BaseViewContract

class RecipeListContract {

    interface View : BaseViewContract {
        fun setupView()
        fun provideSearchConfig(): RecipeConfig
        fun showFooter()
        fun appendRecipes(recipes: List<Recipe>)
        fun showDetail(recipeId: String)
        fun showFilterBottomSheet(
            cuisines: List<String>,
            selectedCuisine: String?
        )

        fun hideFooterLoading()
    }

    interface Presenter : BasePresenterContract {
        fun handleListItemClicked(selectedRecipeId: String)
        fun handleRefresh()
        fun handleLoadMore(lastPosition: Int)
        fun handleFilterActionClicked()
        fun handleFilterItemSelected(cuisine: String)
    }
}
